api/src/main.rs

```diff
@@ -35,6 +36,54 @@ async fn current_weather() -> impl Responder {
     HttpResponse::Ok().json(weather)
 }
 
+#[get("/weather")]
+async fn weather_fn(req : HttpRequest) -> impl Responder {
+    let query_str = req.query_string(); 
+    let qs = QString::from(query_str); // deconstruct the query string for parameters
+    let q_param_get = qs.get("q"); // REQUIRED q query param (zip .. TZ .. lat,long)
+    let d_param_get = qs.get("d"); // d days forecast param, accepts number of days
+
+    // If no q param was given, send a bad request response
+    if q_param_get.is_none() {
+        return HttpResponse::BadRequest().body("Invalid q param specificed.")
+
+    // Otherwise parse the request
+    } else {
+
+        // Get q & d params
+        let q_param = q_param_get.unwrap();
```

Comment:
We likely should be using `match` blocks so we can be sure that we're not unwrapping something and falling over when that fails.

Reply:
Will revise in v2.